From c68ba86adab6629524c71d3cfb8b43b95236d932 Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Tue, 24 May 2022 12:08:49 +0200
Subject: [PATCH] Patch not-yet-up-to-date fill-stroke spec properties

The "<'background'> with modifications" value for the `fill` and `stroke`
properties is not proper CSS grammar. Dropping the definitions means we'll fall
back on the SVG ones for now. Tracked in:
https://github.com/w3c/csswg-drafts/issues/7285

The patch also fixed a regression since SVG Strokes or SVG2 for the
`stroke-width`, `stroke-dasharray` and `stroke-dashoffset` properties that
should accept unitless values. Tracked in:
https://github.com/w3c/csswg-drafts/issues/3057
---
 ed/css/fill-stroke.json | 36 +++---------------------------------
 1 file changed, 3 insertions(+), 33 deletions(-)

diff --git a/ed/css/fill-stroke.json b/ed/css/fill-stroke.json
index afff13433..557bf0e53 100644
--- a/ed/css/fill-stroke.json
+++ b/ed/css/fill-stroke.json
@@ -132,21 +132,6 @@
         "fillRepeat"
       ]
     },
-    "fill": {
-      "name": "fill",
-      "value": "<'background'> with modifications",
-      "initial": "See individual properties",
-      "appliesTo": "See individual properties",
-      "inherited": "See individual properties",
-      "percentages": "N/A",
-      "computedValue": "See individual properties",
-      "canonicalOrder": "per grammar",
-      "media": "visual",
-      "animatable": "See individual properties",
-      "styleDeclaration": [
-        "fill"
-      ]
-    },
     "fill-opacity": {
       "name": "fill-opacity",
       "value": "<'opacity'>",
@@ -165,7 +150,7 @@
     },
     "stroke-width": {
       "name": "stroke-width",
-      "value": "<length-percentage>#",
+      "value": "[<length-percentage> | <number>]#",
       "initial": "1px",
       "appliesTo": "text and SVG shapes",
       "inherited": "yes",
@@ -261,7 +246,7 @@
     },
     "stroke-dasharray": {
       "name": "stroke-dasharray",
-      "value": "none | <length-percentage>+#",
+      "value": "none | [<length-percentage> | <number>]+#",
       "initial": "none",
       "appliesTo": "text and SVG shapes",
       "inherited": "yes",
@@ -277,7 +262,7 @@
     },
     "stroke-dashoffset": {
       "name": "stroke-dashoffset",
-      "value": "<length-percentage>",
+      "value": "<length-percentage> | <number>",
       "initial": "0",
       "appliesTo": "text and SVG shapes",
       "inherited": "yes",
@@ -419,21 +404,6 @@
         "strokeRepeat"
       ]
     },
-    "stroke": {
-      "name": "stroke",
-      "value": "<'background'> with modifications",
-      "initial": "See individual properties",
-      "appliesTo": "See individual properties",
-      "inherited": "See individual properties",
-      "percentages": "N/A",
-      "computedValue": "See individual properties",
-      "canonicalOrder": "per grammar",
-      "media": "visual",
-      "animatable": "See individual properties",
-      "styleDeclaration": [
-        "stroke"
-      ]
-    },
     "stroke-opacity": {
       "name": "stroke-opacity",
       "value": "<'opacity'>",
-- 
2.36.0.windows.1

